plugin/kubernetes/controller.go

```diff
@@ -365,21 +365,21 @@ func (dns *dnsControl) PodIndex(ip string) (pods []*api.Pod) {
 	return pods
 }
 
-func (dns *dnsControl) SvcIndex(idx string) (svcs []*api.Service) {
+func (dns *dnsControl) SvcIndex(key string) (svcs []*api.Service) {
 	if dns.svcLister == nil {
```

Comment:
these `== nil` check where not applied for all index functions, I think they add nothing and can be removed.

Reply:
`dns.epLister` and  `dns.podLister` and can be nil, depending on kubernetes plugin options.
`dns.svcLister` can never be nil, so we can remove the check here.